pkgs/development/libraries/science/math/tensorrt/extension.nix

```diff
@@ -18,14 +19,20 @@ final: prev: let
     # Return the first file that is supported. In practice there should only ever be one anyway.
     supportedFile = files: findFirst isSupported null files;
     # Supported versions with versions as keys and file as value
-    supportedVersions = filterAttrs (version: file: file !=null ) (mapAttrs (version: files: supportedFile files) tensorRTVersions);
+    supportedVersions = filterAttrs (version: file: file != null) (mapAttrs (version: files: supportedFile files) tensorRTVersions);
     # Compute versioned attribute name to be used in this package set
     computeName = version: "tensorrt_${toUnderscore version}";
     # Add all supported builds as attributes
-    allBuilds = mapAttrs' (version: file: nameValuePair (computeName version) (buildTensorRTPackage (removeAttrs file ["fileVersionCuda"]))) supportedVersions;
+    allBuilds = mapAttrs' (version: file: nameValuePair (computeName version) (buildTensorRTPackage (removeAttrs file [ "fileVersionCuda" ]))) supportedVersions;
     # Set the default attributes, e.g. tensorrt = tensorrt_8_4;
-    defaultBuild = { "tensorrt" = allBuilds.${computeName tensorRTDefaultVersion}; };
-  in allBuilds // defaultBuild;
+    defaultBuildName = computeName tensorRTDefaultVersion;
+    defaultBuild =
```

Comment:
I'm only touching this because of [eval errors](https://hercules-ci.com/github/SomeoneSerge/nixpkgs-unfree/jobs/2163).
But I'm also really overwhelmed reading this piece

Reply:
Fix seems good to me.

We create an attribute `tensorr_${version}` for every available `tensorrt` version. Then we add a `tensorrt` attribute (so without version`, however, for cudatoolklit 11.8 there is no compatible `tensorrt`, and hence we cannot add a default.

In the future we might need a fix like this for the other extensions as well.